src/fragments/lib/geo/ios/search/30_location_search_map.mdx

```diff
@@ -45,6 +48,55 @@ struct MyMapView: View {
 }
 ```
 
+</Block>
+
+<Block name="Developer Preview">
+
+```swift
+import SwiftUI
+import AmplifyMapLibreUI
+import AmplifyMapLibreAdapter
+import Amplify
+
+struct MyMapView: View {
+
+    @StateObject private var mapState = AMLMapViewState()
+    @State private var searchText = ""
+    @State private var displayState: AMLSearchBar.DisplayState = .map
+
+    var body: some View {
+        ZStack(alignment: .top) {
+            AMLMapView(mapState: mapState)
+                .edgesIgnoringSafeArea(.all)
+
+            AMLSearchBar(
+                text: $searchText,
+                displayState: $displayState,
+                onEditing: { },
+                onCommit: search,
+                onCancel: { mapState.features = [] }
+            )
+            .padding()
+        }
+    }
+
+    private func search() {
+        let searchArea = Geo.SearchArea.near(mapState.center)
+        let searchOptions = Geo.SearchForTextOptions(area: searchArea)
+        Task {
+            do {
+                let places = try await Amplify.Geo.search(for: searchText, options: searchOptions)
+                self.mapState.features = AmplifyMapLibre.createFeatures(places)
```

Comment:
This needs to be on the main thread (MainActor) because it will trigger UI updates.  
```suggestion
                await MainActor.run {
                    self.mapState.features = AmplifyMapLibre.createFeatures(places)
                }
```